module/zfs/vdev_mirror.c

```diff
@@ -252,16 +252,21 @@ vdev_mirror_map_init(zio_t *zio)
 		dva_t *dva = zio->io_bp->blk_dva;
 		spa_t *spa = zio->io_spa;
 
-		mm = vdev_mirror_map_alloc(BP_GET_NDVAS(zio->io_bp), B_FALSE,
-		    B_TRUE);
+		c = BP_GET_NDVAS(zio->io_bp);
+		if (zio->io_mirror_child_hint != -1)
+			c = zio->io_mirror_child_hint % c + 1;
```

Comment:
the `vd==NULL` case shouldn't be changed.

Reply:
OK, I wasn't sure about that.